packages/cpt-ui/src/context/AccessProvider.tsx

```diff
@@ -108,6 +108,19 @@ export const AccessProvider = ({ children }: { children: ReactNode }) => {
 
       const userInfo: TrackerUserInfo = data.userInfo
 
+      if (userInfo) {
+        if (userInfo.roles_with_access) {
+          setRolesWithAccess(userInfo.roles_with_access);
+        } else {
+          const storedRolesWithAccess = localStorage.getItem('rolesWithAccess');
```

Comment:
This solution works just fine, but I do have a suggestion (ignore this if I'm off the mark).

We have the `localStorageState` which can restore the state of something from the cache on a reload, which should also work here. Though, since you have extra logic and it's 5pm, I'm not 100% certain the logic is translatable so I may be wrong.